task_2.py

```diff
@@ -0,0 +1,140 @@
+from pprint import pprint
+from collections import defaultdict
+from collections import namedtuple
+from datetime import datetime, timedelta
+import requests
+import csv
+
+
+class Movie_data:
+    """Class to work with api data(movies) """
+
+    def __init__(self, pages_numb):
+        self.pages = pages_numb
+        self.url = 'https://api.themoviedb.org/3/discover/movie?include_adult=false&include_video=false&sort_by=popularity.desc&page={}'
+        self.films_data = []
+
+        self.headers = {
+            'accept': 'application/json',
+            'Authorization': 'Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiIzMTI3NGFmYTRlNTUyMjRjYzRlN2Q0NmNlMTNkOTZjOSIsInN1YiI6IjVkNmZhMWZmNzdjMDFmMDAxMDU5NzQ4OSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.lbpgyXlOXwrbY0mUmP-zQpNAMCw_h-oaudAJB6Cn5c8'
+        }
+        self.fetch_data()
+        self.genres = self.fetch_genres()
+
+    def fetch_data(self):
+        for page in range(1, self.pages + 1):
+            url = self.url.format(page)
+            response = requests.get(url, headers=self.headers)
+            if response.status_code == 200:
+                self.films_data.extend(response.json()['results'])
+
+    def fetch_genres(self):
+        url = 'https://api.themoviedb.org/3/genre/movie/list?language=en'
+        response = requests.get(url, headers=self.headers)
+        if response.status_code == 200:
+            return {genre['id']: genre['name'] for genre in response.json()['genres']}
+
+    def give_all_data(self):
+        return self.films_data
+
+    def give_data_from_pages(self, pages):
+        return self.films_data[:pages]
+
+    def give_films_by_index(self, start, end, step):
+        return self.films_data[start:end:step]
+
+    def give_most_popular_film(self):
+        return max(self.films_data, key=lambda film: film['popularity'])['title']
+
+    def give_movie_by_keywords(self, keywords):
+        titles_names = [movie['title'] for movie in self.films_data if any(keywords in movie['overview'] for keyword in keywords)]
+        # titles_names = []
+        # for movie in self.films_data:
+        #     if any(keywords in movie['overview'] for keyword in keywords):
+        #         titles_names.append(movie['title'])
+        return titles_names
+
+    def give_unique_genres(self):
+        return frozenset(genre for genre in self.genres.values())
+
+    def delete_movie_by_genre(self, genre):
+        genre_id = [key for key in self.genres if self.genres[key] == genre]
+        self.films_data = list(filter(lambda x: genre_id[0] not in x['genre_ids'], self.films_data))
+        return 'movies deleted'
+
+    def give_most_popular_genre(self):
+        genre_c = [genre_id for movie in self.films_data for genre_id in movie['genre_ids']]
+        # for movie in self.films_data:
+        #     for genre_id in movie['genre_ids']:
+        #         genre_c.append(genre_id)
+        max_v = max(genre_c)
+        return self.genres[max_v]
```

Comment:
firstly, there is a mistake. now you just find the biggest id number. not a count of occurrences in the data.

secondly, better to write it in one line